moveit_setup_assistant/src/widgets/configuration_files_widget.cpp

```diff
@@ -884,6 +884,13 @@ void ConfigurationFilesWidget::changeCheckedState(QListWidgetItem* item)
 
   // Enable/disable file
   gen_files_[index].generate_ = generate;
+
+  // Update GAZEBO_URDF_LOAD_ATTRIBUTE to load URDF file from config folder or to regenerate by xacro
+  if (gen_files_[index].write_on_changes == MoveItConfigData::SIMULATION)
+  {
+    config_data_->save_gazebo_urdf_ = generate;
+    loadGenFiles();
```

Comment:
I gues you reload the to-be-generated files because `save_gazebo_urdf_` flag has changed?

Reply:
Exactly, but actually only the depending files have to be regenerated or generated first, when all flags are certain.